invokeai/app/util/step_callback.py

```diff
@@ -20,12 +21,18 @@ def fast_latents_step_callback(
 
     dataURL = image_to_dataURL(image, image_format="JPEG")
 
+    graph_execution_state = context.services.graph_execution_manager.get(context.graph_execution_state_id)
```

Comment:
This is pretty expensive to perform every step (since it could be a database lookup), and the client should already have the mapping from execution node id to source node id. Handle this client-side, and just send the current execution node back (if not just the execution node id, since the client should already have the full execution node, so this is a waste of bandwidth).

Reply:
At this point, the client does not have the mappings.

Are the nodes ids mapped by the time we send `invocation_started` event? If so, we could send the prepared execution state on the that event and never worry about sending the source node id.

Why this is here: We need the source node id on every event so we can selectively handle events (that is, to ignore events we don't care about).

Reply:
refactored this, addressing these issues